x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/blocklist_validator.ts

```diff
@@ -0,0 +1,284 @@
+ */
+const EntriesSchema = schema.conditional(
+  schema.contextRef('os'),
+  OperatingSystem.WINDOWS,
+  schema.arrayOf(WindowsEntrySchema, entriesSchemaOptions),
+  schema.conditional(
+    schema.contextRef('os'),
+    OperatingSystem.LINUX,
+    schema.arrayOf(LinuxEntrySchema, entriesSchemaOptions),
+    schema.arrayOf(MacEntrySchema, entriesSchemaOptions)
+  )
+);
+
+/**
+ * Schema to validate Blocklist data for create and update.
+ * When called, it must be given an `context` with a `os` property set
+ *
+ * @example
+ *
+ * BlocklistDataSchema.validate(item, { os: 'windows' });
+ */
+const BlocklistDataSchema = schema.object(
+  {
+    entries: EntriesSchema,
+  },
+
+  // Because we are only validating some fields from the Exception Item, we set `unknowns` to `ignore` here
+  { unknowns: 'ignore' }
+);
+
+export class BlocklistValidator extends BaseValidator {
+  static isBlocklist(item: { listId: string }): boolean {
+    return item.listId === ENDPOINT_BLOCKLISTS_LIST_ID;
+  }
+
+  async validatePreCreateItem(
+    item: CreateExceptionListItemOptions
+  ): Promise<CreateExceptionListItemOptions> {
+    await this.validateCanManageEndpointArtifacts();
+    await this.validateBlocklistData(item);
+    await this.validateCanCreateByPolicyArtifacts(item);
+    await this.validateByPolicyItem(item);
+
+    return item;
+  }
+
+  async validatePreDeleteItem(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreGetOneItem(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreMultiListFind(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreExport(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreSingleListFind(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreGetListSummary(): Promise<void> {
+    await this.validateCanManageEndpointArtifacts();
+  }
+
+  async validatePreUpdateItem(
+    _updatedItem: UpdateExceptionListItemOptions,
+    currentItem: ExceptionListItemSchema
+  ): Promise<UpdateExceptionListItemOptions> {
+    const updatedItem = _updatedItem as ExceptionItemLikeOptions;
+
+    await this.validateCanManageEndpointArtifacts();
+    await this.validateBlocklistData(updatedItem);
+
+    try {
+      await this.validateCanCreateByPolicyArtifacts(updatedItem);
+    } catch (noByPolicyAuthzError) {
+      // Not allowed to create/update by policy data. Validate that the effective scope of the item
+      // remained unchanged with this update or was set to `global` (only allowed update). If not,
+      // then throw the validation error that was catch'ed
+      if (this.wasByPolicyEffectScopeChanged(updatedItem, currentItem)) {
+        throw noByPolicyAuthzError;
+      }
+    }
+
+    await this.validateByPolicyItem(updatedItem);
+
+    return updatedItem as UpdateExceptionListItemOptions;
```

Comment:
You could just return `_updatedItem` here and won't need to cast type again.

Reply:
agree with Ash. It could also avoid unforeseen conditions where `updatedItem` might have actually be mutated or changed